### PR TITLE
Clean up StartNewLvl

### PR DESCRIPTION
**Pull request by AJenbo**
Bin exact

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -2228,51 +2228,41 @@ void __fastcall InitLevelChange(int pnum)
...
-		if ( v3 != WM_DIABTOWNWARP )
-		{
-			if ( v3 != WM_DIABTWARPUP )
-			{
-				if ( v3 == WM_DIABRETOWN )
-					goto LABEL_11;
-				goto LABEL_10;
-			}
+
+	switch ( fom ) {
```
Comment (mewmew) [inline Source/player.cpp]: Would

```
		case WM_DIABNEXTLVL:
		case WM_DIABPREVLVL:
		case WM_DIABRTNLVL:
		case WM_DIABTOWNWARP:
			plr[pnum].plrlevel = lvl;
			break;
```

Still produce bin exact? If so, the two switch cases can be merged into one, as the case body of WM_DIABNEXTLVL, WM_DIABPREVLVL is the same as the case body of WM_DIABRTNLVL, WM_DIABTOWNWARP

File: Source/player.cpp [hunk 2]
```diff
@@ -2228,51 +2228,41 @@ void __fastcall InitLevelChange(int pnum)
...
-		v6 = ghMainWnd;
-		plr[v5]._pmode = PM_NEWLVL;
-		plr[v5]._pInvincible = 1;
-		PostMessageA(v6, v3, 0, 0);
-		if ( (unsigned char)gbMaxPlayers > 1u )
-			NetSendCmdParam2(1u, CMD_NEWLVL, v3, lvl);
+
+	if ( pnum == myplr ) {
+		plr[pnum]._pmode = PM_NEWLVL;
+		plr[pnum]._pInvincible = 1;
```
Comment (mewmew) [inline Source/player.cpp]: Is _pInvincible a boolean? If so use TRUE

File: Source/player.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/player.cpp]: no it changes it :(

Comment (mewmew) [reply Source/player.cpp]: That's ok. Keep it as is then

File: Source/player.cpp (hunk 2 above)
Comment (seritools) [reply Source/player.cpp]: @AJenbo nah, many fields in the PlayerStruct are size optimized. You can easily see which one is right for `plr[pnum]._pInvincible = 1;` by looking at the pointed size, `dword/word/byte ptr` and the corresponding register the value came from (if any), `eax/ax/al`.

**Pull request merged**